helm-v3/deisgn-proposal.md

```diff
@@ -58,8 +58,9 @@ In an effort to mirror the construction of Kubernetes manifests, the new Chart.y
 apiVersion: helm.sh/v3
 kind: Chart
 metadata:
-  name: myChart
+  name: myChart-0.1.0
```

Comment:
Ick. For three reasons:

1. There is a `version` property and the version in the name.
1. How many people would expect the name to include a version? It's going to make people have to think.
1. If we are working with the Application CRD, why does the Chart.yaml need to be install-able? What's the use case?

Reply:
I agree with Matt that this will be confusing. With the Application CRD and Release/ReleaseVersion CRDs, I also think that the Chart.yaml doesn't need to be installable. IMO Chart.yaml should be used for chart repo metadata (index.yaml) and used to generate Application, Release and ReleaseVersion objects.

Reply:
We don't really know if we are going to use the Application CRD. It's still too much in flux.

But if we do, there's probably no compelling reason to turn the Chart.yaml into a resource-like thing anyway.

The problem is that in the present spec, there can only be one such application/version per namespace, which is not at all the desired behavior. And the only way to get around that is to change the `metadata.name` field.

Reply:
> We don't really know if we are going to use the Application CRD. It's still too much in flux.

We should figure out the criteria on go/no-go soon. At least for first pass on direction. If we don't go the Application CRD and everyone else does we end up shifting expectations onto chart developers to create this if they want to do things like display their apps in dashboard.

> The problem is that in the present spec, there can only be one such application/version per namespace

I'm not sure why a name needs to have a unique name/version. What if, in the same namespace, I deploy multiple wordpress charts at the same version? Uniqueness is gone for a name in a namespace in this model.

What's the underlying use case?

Reply:
Unique name is imposed by Kubernetes.

So the bottom line to me is:

- If we are modeling a Chart.yaml as something installable, then we have to design it to be actually installable.
- Otherwise, can we just drop the "let's pretend this is a Kubernetes resource" crap and stick with our simpler YAML format from Helm 2?

Reply:
See #12, which is probably the better course of action.